OpenRA.Mods.Common/Activities/Air/FlyAttack.cs

```diff
@@ -111,11 +111,17 @@ public override bool Tick(Actor self)
 				return true;
 			}
 
-			// If all valid weapons have depleted their ammo and Rearmable trait exists, return to RearmActor to reload and then resume the activity
+			// If all valid weapons have depleted their ammo and Rearmable trait exists, return to RearmActor to reload
 			if (rearmable != null && !useLastVisibleTarget && attackAircraft.Armaments.All(x => x.IsTraitPaused || !x.Weapon.IsValidAgainst(target, self.World, self)))
 			{
-				QueueChild(new ReturnToBase(self));
-				return aircraft.Info.AbortOnResupply;
+				// End this activity after reload has been queued if AbortOnResupply is true,
```

Comment:
This is what the `return attackAircraft.Info.AbortOnResupply;` should already be doing. Are you sure these changes are still needed after the rebase?

Reply:
No, but I also wasn't sure what happens to the child when the parent activity ends. It seems to work as intended though, so I'll drop my change.